assets/js/script.js

```diff
@@ -0,0 +1,429 @@
+      if (characterX < 0) characterX = 560
+      character.style.top = Math.round(characterY / 40) * 40 + 'px'
+      character.style.left = characterX - ghostSpeed + 'px'
+    }
+    if (character.direction === 'right') {
+      character.className = ''
+      if (characterX > 560) characterX = 0
+      character.style.top = Math.round(characterY / 40) * 40 + 'px'
+      character.style.left = characterX + ghostSpeed + 'px'
+    }
+    if (hitWall(character)) {
+      character.style.left = charOriginalLeft
+      character.style.top = charOriginalTop
+      ghostHitWallNewDir(character)
+    }
+  }
+
+  // find new direction when ghost hit wall
+  function ghostHitWallNewDir (character) {
+    var oppDirection
+    var newDirection
+
+    if (character.direction === 'up') oppDirection = 'down'
+    if (character.direction === 'down') oppDirection = 'up'
+    if (character.direction === 'left') oppDirection = 'right'
+    if (character.direction === 'right') oppDirection = 'left'
+
+    do {
+      newDirection = randomGhostDirection()
+    } while (newDirection === oppDirection)
+
+    character.direction = newDirection
+  }
+
+  // random ghost direction
+  function randomGhostDirection () {
+    var ghostDirection = ['up', 'down', 'left', 'right']
+    var direction = ghostDirection[Math.floor(Math.random() * (4))]
+    return direction
+  }
+
+  // check score
+  function checkScore () {
+    for (var i = 0; i < dots.length; i++) {
+      if (collision(dots[i], pacman)) {
+        if (!scoretracker.includes(dots[i])) {
+          dots[i].style.display = 'none'
+          scoretracker.push(dots[i])
+          scores.textContent = scoretracker.length + '/92'
+        }
+      }
+    }
+    if (scoretracker.length === 92) {
+      gameOver('win')
+    }
+  }
+
+  // game over
+  function gameOver (result) {
+    if (result === 'win') {
+      clearInterval(loopTimer)
+      win.style.display = ''
+      startButton.style.display = ''
+    } else if (result === 'lose') {
+      if (lives === 1) {
+        clearInterval(loopTimer)
+        document.querySelector('#life1').style.display = 'none'
+        lose.style.display = ''
+        startButton.style.display = ''
+      } else {
+        document.querySelector('#life' + lives).style.display = 'none'
+        lives--
+        clearInterval(loopTimer)
+        interval.style.display = ''
+        interval.textContent = 'GET READY!'
+        interval.style.background = 'rgb(134, 231, 196)'
+        pacman.style.top = '440px'
+        pacman.style.left = '280px'
+        blueGhost.style.top = '40px'
+        blueGhost.style.left = '40px'
+        orangeGhost.style.top = '40px'
+        orangeGhost.style.left = '520px'
+        pinkGhost.style.top = '440px'
+        pinkGhost.style.left = '40px'
+        redGhost.style.top = '440px'
+        redGhost.style.left = '520px'
+        upKeyDown = false
+        downKeyDown = false
+        leftKeyDown = false
+        rightKeyDown = false
```

Comment:
is there a way to not repeat so much of your upKeyDown, downKeyDown, leftKeyDown, and rightKeyDown?